lib/serviceClient.ts

```diff
@@ -189,60 +191,127 @@ export class ServiceClient {
     httpRequest.method = operationSpec.httpMethod;
     httpRequest.operationSpec = operationSpec;
 
+    const requestUrl: URLBuilder = URLBuilder.parse(operationSpec.baseUrl);
+    if (operationSpec.path) {
+      requestUrl.setPath(operationSpec.path);
+    }
+    if (operationSpec.urlParameters && operationSpec.urlParameters.length > 0) {
+      for (const urlParameter of operationSpec.urlParameters) {
+        let urlParameterValue: string = operationArguments.arguments[urlParameter.parameterName];
+        if (urlParameter.type != undefined) {
+          urlParameterValue = serializeParameterValue(urlParameterValue, urlParameter.type, this._serializer, urlParameter.parameterName);
+        }
+        if (!urlParameter.skipEncoding) {
+          urlParameterValue = encodeURIComponent(urlParameterValue);
+        }
+        requestUrl.replaceAll(`{${urlParameter.urlParameterName || urlParameter.parameterName}}`, urlParameterValue);
+      }
+    }
+    if (operationSpec.queryParameters && operationSpec.queryParameters.length > 0) {
+      for (const queryParameter of operationSpec.queryParameters) {
+        let queryParameterValue: any = operationArguments.arguments[queryParameter.parameterName];
+        if (queryParameterValue != undefined) {
+          if (queryParameter.type != undefined) {
+            queryParameterValue = serializeParameterValue(queryParameterValue, queryParameter.type, this._serializer, queryParameter.parameterName);
+          }
+          if (queryParameter.collectionFormat != undefined) {
+            if (queryParameter.collectionFormat === QueryCollectionFormat.Multi) {
+              if (queryParameterValue.length === 0) {
+                queryParameterValue = "";
+              } else {
+                for (const item of queryParameterValue) {
+                  queryParameterValue = (item == undefined ? "" : item.toString());
+                }
+              }
+            } else {
+              let queryParameterValueSeparator: string;
+              switch (queryParameter.collectionFormat) {
```

Comment:
Can this be baked into the QueryCollectionFormat enum?

Reply:
I can't make the enum a string enum (if that's what you're asking), but I can move this logic into the same file.

Reply:
Yeah, I think that's adequate. That's because you want to fail for multi, basically?

Reply:
Correct.